apps/election-manager/src/components/ExportFinalResultsModal.tsx

```diff
@@ -45,17 +42,32 @@ const ExportFinalResultsModal: React.FC<Props> = ({
     usbDriveStatus,
     castVoteRecordFiles,
     electionDefinition,
+    fullElectionExternalTally,
   } = useContext(AppContext)
 
   const [currentState, setCurrentState] = useState(ModalState.INIT)
   const [errorMessage, setErrorMessage] = useState('')
 
   const [savedFilename, setSavedFilename] = useState('')
+  const [includeExternalFile, setIncludeExternalFile] = useState(true)
+  const [defaultFilename, setDefaultFilename] = useState('')
+
+  const isTestMode = castVoteRecordFiles?.fileMode === 'test'
+
+  useEffect(() => {
+    setDefaultFilename(
+      generateFinalExportDefaultFilename(
+        isTestMode,
+        electionDefinition!.election
+      )
+    )
+  }, [isOpen])
```

Comment:
I don't want it to create a new name every time you toggle the segmented button for including the SEMs data, since the timestamp on the file is shown in the dialog and is to the second it's kind of distracting/weird for that to be changing.

Reply:
Wouldn't this work without the `useEffect`?

```ts
const [defaultFilename, setDefaultFilename] = useState(      
  generateFinalExportDefaultFilename(
    isTestMode,
    electionDefinition!.election
  )
)
```

Reply:
Since this is a model it gets rendered whenever the surrounding tallyscreen page is loaded, and I do want to regenerate the filename when the modal is closed and then reopened, which is why I'm depending on isOpen changing in the useEffect

Reply:
Interesting issue. Just seems like useEffect isn't the "right" way to do this. I would have also suggested a solution similar to what Brian suggested. Maybe add a comment in the useEffect block to clarify what the useEffect is for?